src/hotspot/share/prims/jvm.cpp

```diff
@@ -2913,6 +2913,11 @@ JVM_ENTRY(void, JVM_StartThread(JNIEnv* env, jobject jthread))
         JVMTI_RESOURCE_EXHAUSTED_OOM_ERROR | JVMTI_RESOURCE_EXHAUSTED_THREADS,
         os::native_thread_creation_failed_msg());
     }
+
+    // SapMachine 2021-05-21: All ...OnOutOfMemoryError switches should work for
+    //  thread creation failures too.
+    report_java_out_of_memory(os::native_thread_creation_failed_msg());
```

Comment:
In case of an OOM from the heap the report_java_out_of_memory() method is called before the JVMTI event is optionally generated. Is there a good reason to do this differently here?

Reply:
No, as I wrote I was torn here. The question is whether we want things like CF jvmkill to have first dibs on OOM handling. Well maybe not :) I'll change this and do it before JVMTI.